crates/bevy_asset/src/lib.rs

```diff
@@ -1,3 +1,106 @@
+//! Note that this method does not attempt to reload the asset if it has already been loaded: as long as the asset hasn't been dropped,
+//! calling [`AssetServer::load`] on the same path will return the same handle.
+//! The handle that's returned will be used to instantiate various [`Component`](bevy_ecs::prelude::Component)s that require asset data to function,
+//! which will then be spawned into the world as part of an entity.
+//!
+//! To avoid assets "popping" into existence, you may want to check that all of the required assets are loaded before transitioning to a new scene.
+//! This can be done by checking the [`LoadState`] of the asset handle using [`AssetServer::is_loaded_with_dependencies`],
+//! which will be [`LoadState::Loaded`] when the asset is ready to use.
+//! Keep track of what you're waiting on using a [`HashSet`] or similar data structure,
+//! poll in your update loop, and transition to the new scene when all assets are loaded.
+//! Bevy's built-in states system can be very helpful for this!
+//!
+//! If we later want to manipulate this asset data (such as for displaying a death animation), we have three options:
+//!
+//! 1. Despawn the entity and spawn a new one with the new asset data.
+//! 2. Change what the handle is pointing to.
+//! 3. Use the [`Assets`] collection to directly modify the asset data.
+//!
+//! The first option is the simplest, but can be slow if done frequently,
+//! and can lead to frustrating bugs as references to the old entity (such as what is targeting it) and other data on the entity are lost.
+//! Generally, this isn't a great strategy.
+//!
+//! The second option is the most common: just query for the component that holds the handle, and mutate it, pointing to the new asset.
+//!
+//! The third option has different semantics: rather than modifying the asset data for a single entity, it modifies the asset data for *all* entities using this handle.
+//! While this might be what you want, it generally isn't!
+//!
+//! ## Handles and reference counting
+//!
+//! [`Handle`] (or their untyped counterpart [`UntypedHandle`]) are used to reference assets in the [`Assets`] collection,
+//! and are the primary way to interact with assets in Bevy.
+//! As a user, you'll be working with handles a lot!
+//!
+//! The most important thing to know about handles is that they are reference counted: when you clone a handle, you're incrementing a reference count.
+//! When the object holding the handle is dropped (generally because an entity was despawned), the reference count is decremented.
+//! When the reference count hits zero, the asset it references is removed from the [`Assets`] collection.
+//! To avoid incrementing the reference count, you can use the [`Handle::clone_weak`] method, which is marginally faster.
+//!
+//! This reference counting is a simple, laregely automatic way to avoid holding onto memory for game objects that are no longer in use.
```

Comment:
```suggestion
//! This reference counting is a simple, largely automatic way to avoid holding onto memory for game objects that are no longer in use.
```